File: 2025-2/sesion-de-clase-02/Java/ejercicio01/src/main/java/pe/edu/pucp/inf30/ejercicio01/Alumno.java
package pe.edu.pucp.inf30.ejercicio01;

/**
 *
 * @author eric
 */
public class Alumno extends Persona implements IPracticable {
    private TipoActividad actividad;
    
    public Alumno() {
        System.out.println("Alumno: Constructor");
        this.actividad = TipoActividad.Ninguna;
    }
    
    public TipoActividad getActividad() {
        return this.actividad;
    }
    
    public void setTipoActividad(TipoActividad actividad) {
        this.actividad = actividad;
    }
    
    Alumno(AlumnoBuilder builder) {
        super(builder);
        this.actividad = builder.getActividad();
    }
    
    @Override
    public String toString() {
        return String.format("[Alumno] Nombre: %s, Edad: %d, "
                + "Actividad: %s", 
                this.getNombre(), this.getEdad(), this.actividad);
    }
    
    @Override
    public void imprimir() {
        System.out.println(this);
    }

    @Override
    public void practicar() {
        System.out.println("El alumno esta practicando: " + this.actividad);
    }
    
    public static class AlumnoBuilder extends Persona.PersonaBuilder {
        private TipoActividad actividad = TipoActividad.Ninguna;
        
        public TipoActividad getActividad() {
            return this.actividad;
        }

        @Override
        public AlumnoBuilder nombre(String nombre) {
            this.nombre = nombre;
            return this;
        }

        @Override
        public AlumnoBuilder edad(int edad) {
            this.edad = edad;
            return this;
        }
        
        public AlumnoBuilder actividad(TipoActividad actividad) {
            this.actividad = actividad;
            return this;
        }
        
        @Override
        public Alumno build() {
            return new Alumno(this);
        }
    }
}

File: 2025-2/sesion-de-clase-02/dotNET/Ejercicio01/Alumno.cs
using System;

namespace Ejercicio01 {
    public class Alumno : Persona, IPracticable {
        public TipoActividad Actividad { get; set; }

        public Alumno() : base() {
            this.Actividad = TipoActividad.Ninguna;
            Console.WriteLine("Constructor: Alumno");
        }

        Alumno(AlumnoBuilder builder) : base(builder) {
            this.Actividad = builder.Actividad;
        }

        public override string ToString() {
            return string.Format("[Alumno] Nombre: {0}, Edad: {1}, Actividad: {2}", this.Nombre, this.Edad, this.Actividad);
        }

        public override void Imprimir() {
            Console.WriteLine(this); // Llama al metodo ToString()
        }

        public void Practicar() {
            Console.WriteLine("El alumno está practicando: " + this.Actividad);
        }

        public class AlumnoBuilder : Persona.PersonaBuilder {
            public TipoActividad Actividad { get; private set; }

            public AlumnoBuilder ConActividad(TipoActividad actividad) {
                this.Actividad = actividad;
                return this;
            }

            public override Persona Build() {
                return new Alumno(this);
            }
        }
    }
}
